import "common.proto";
//任务
message Task1
{
    required int32 tid = 1; //任务ID
    repeated TaskCondition condition = 2; //任务当前值,如果状态是完成,不法这个值
    required int32 status = 3;//当前状态
}

message TaskCondition
{
    required int32 condition_no = 1;
    optional int32 current = 2;
    optional int32 state = 3;  // 0未完成 1 完成
}

//获取任务列表
message TaskInfoRequest
{
    required int32 sort = 1; //任务分类
}

//获取任务列表
message TaskInfoResponse
{
    repeated Task1 tasks = 1; //任务列表
}

//请求领取奖励
message TaskRewardRequest
{
	required int32 tid = 1; //任务ID
}

//领奖返回
message TaskRewardResponse
{
	optional int32 tid = 1; //任务ID
	optional GameResourcesResponse gain = 2; //通用奖励
    required CommonResponse res = 3;
}
